### PR TITLE
[MHLO] Support AtenMaskedFillScalar

### PR DESCRIPTION
**Pull request by ZihengJiang**

### CODE REVIEW

File: lib/Conversion/TorchToMhlo/Basic.cpp [hunk 1]
```diff
@@ -1292,6 +1292,51 @@ LogicalResult ConvertAtenOp<AtenArangeStartStepOp>::matchAndRewrite(
   return success();
 }
 
+// AtenMaskedFillScalarOp
+template <>
+LogicalResult ConvertAtenOp<AtenMaskedFillScalarOp>::matchAndRewrite(
```
Comment (tanyokwok) [inline lib/Conversion/TorchToMhlo/Basic.cpp:1297]: I think we can decompose AtenMaskedFillScalarOp into `AtenWhereScalarSelfOp`, which is more straightforward

Comment (tanyokwok) [reply lib/Conversion/TorchToMhlo/Basic.cpp:1297]: And the `AtenWhereScalarSelfOp` can be convert into `mhlo::BroadcastSelectOp`

**Comment by ZihengJiang** [comment]
The CI is failed due to TOSA failed the `MaskedFillScalarIntValueModule_basic` test, but this test is actually not added into the TOSA_PASS_SET in xfail_sets. Any idea? @ramiro050

**Comment by ramiro050** [comment]
The test seems to be passing

```
Unexpected outcome summary:

****** Unexpectedly Passed tests - 1 tests
    XPASS - "MaskedFillScalarIntValueModule_basic"
```

Should just be a matter of adding it to the tosa_pass_set

File: lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp [hunk 2]
```diff
@@ -1390,6 +1390,25 @@ class DecomposeAtenWhereScalarSelfOp
 };
 } // namespace
 
+// Decompose aten.masked_fill.Scalar into aten.where.self op.
+namespace {
+class DecomposeAtenMaskedFillScalarOp
```
Comment (ramiro050) [inline lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp:1395]: I think we can remove the linalg lowering for this op now that we have the decomposition

File: lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp [hunk 3]
```diff
@@ -3821,6 +3840,7 @@ class DecomposeComplexOpsPass
     addPatternIfTargetOpIsIllegal<DecomposeAtenWhereScalarOp>(patterns);
     addPatternIfTargetOpIsIllegal<DecomposeAtenWhereScalarOtherOp>(patterns);
     addPatternIfTargetOpIsIllegal<DecomposeAtenWhereScalarSelfOp>(patterns);
+    addPatternIfTargetOpIsIllegal<DecomposeAtenMaskedFillScalarOp>(patterns);
```
Comment (ramiro050) [inline lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp:3843]: You also need to add the op here: https://github.com/llvm/torch-mlir/blob/eb7bac7ca43ef76581f025fb1c0417308a29582a/lib/Dialect/Torch/Transforms/LowerToBackendContract.cpp#L335

**Review by ramiro050 — CHANGES_REQUESTED** [review]

**Review by ramiro050 — APPROVED** [review]
Thanks!

**Pull request merged**